src/standalone/Vegetation/photosynthesis_farquhar.jl

```diff
@@ -0,0 +1,716 @@
+        :fC4 => FT(0.025),
+        :Q10 => FT(2),
+        :s1 => FT(0.3),
+        :s2 => FT(313.15),
+        :s3 => FT(0.2),
+        :s4 => FT(288.15),
+        :s5 => FT(1.3),
+        :s6 => FT(328.15),
+        :E => FT(0.05),
+    )
+    # if is_c3 is a field, is_c3 may contain values between 0.0 and 1.0 after regridding
+    # this deals with that possibility by rounding to the closest int
+    is_c3 = max.(min.(is_c3, FT(1)), FT(0)) # placeholder
+    is_c3 = round.(is_c3)
+    MECH = typeof(is_c3)
+    Vcmax25 = FT.(Vcmax25)
+    VC = typeof(Vcmax25)
+    return FarquharParameters{FT, MECH, VC}(;
+        is_c3,
+        Vcmax25,
+        parameters...,
+        C4_parameters...,
+        kwargs...,
+    )
+end
+
+"""
+    max_electron_transport_farquhar(
+        Vcmax25::FT,
+        ΔHJmax::FT,
+        T::FT,
+        To::FT,
+        R::FT,
+    ) where {FT}
+
+Computes the maximum potential rate of electron transport (`Jmax`),
+in units of mol/m^2/s,
+as a function of Vcmax at 25 °C (`Vcmax25`),
+a constant (`ΔHJmax`), a standard temperature (`To`),
+the unversal gas constant (`R`), and the temperature (`T`).
+
+See Table 11.5 of G. Bonan's textbook,
+Climate Change and Terrestrial Ecosystem Modeling (2019).
+"""
+function max_electron_transport_farquhar(
+    Vcmax25::FT,
+    ΔHJmax::FT,
+    T::FT,
+    To::FT,
+    R::FT,
+) where {FT}
+    Jmax25 = Vcmax25 * FT(exp(1))
+    Jmax = Jmax25 * arrhenius_function(T, To, R, ΔHJmax)
+    return Jmax
+end
+
+"""
+    electron_transport_farquhar(APAR::FT,
+                       Jmax::FT,
+                       θj::FT,
+                       ϕ::FT) where {FT}
+
+Computes the rate of electron transport (`J`),
+in units of mol/m^2/s, as a function of
+the maximum potential rate of electron transport (`Jmax`),
+absorbed photosynthetically active radiation (`APAR`),
+an empirical "curvature parameter" (`θj`; Bonan Eqn 11.21)
+and the quantum yield of photosystem II (`ϕ`).
+
+See Ch 11, G. Bonan's textbook, Climate Change and Terrestrial Ecosystem Modeling (2019).
+"""
+function electron_transport_farquhar(
+    APAR::FT,
+    Jmax::FT,
+    θj::FT,
+    ϕ::FT,
+) where {FT}
+    # Light utilization of APAR
+    IPSII = ϕ * APAR / 2
+    # This is a solution to a quadratic equation
+    # θj *J^2 - (IPSII+Jmax)*J+IPSII*Jmax = 0, Equation 11.21
+    J =
+        (IPSII + Jmax - sqrt((IPSII + Jmax)^2 - 4 * θj * IPSII * Jmax)) /
+        (2 * θj)
+    return J
+end
+
+"""
+    GPP_from_leaf_level_A(A::FT,
+                          K::FT,
+                          LAI::FT,
+                          Ω::FT) where {FT}
+
+Computes the gross canopy photosynthesis (`GPP`) as a function of
+the gross carbon assimilation (`A`) at the leaf level, the extinction coefficient (`K`),
+leaf area index (`LAI`) and the clumping index (`Ω`).
+"""
+function GPP_from_leaf_level_A(A::FT, K::FT, LAI::FT, Ω::FT) where {FT}
+    GPP = A * (1 - exp(-K * LAI * Ω)) / (K * Ω)
+    return GPP
+end
```

Comment:
As discussed, I think this is incorrect for our case.

This is basically making a Beer-Lambert attenuation to account for APAR. 
But we already account for APAR (which is good, better than beer lambert), so that our A (leaf level) represents an average leaf --> so we can just do

GPP = A * LAI

Reply:
I fixed this. since it is such a simple computation, I removed GPP from the cache and introduced a helper function `get_GPP_canopy()` which computes `A*LAI as a diagnostic. in the pmodel it is a bit more complex, so GPP is still in the cache there `get_GPP_canopy()` returns `p....GPP`.